src/framework/entity.js

```diff
@@ -514,34 +514,13 @@ class Entity extends GraphNode {
             this.c[name].enabled = false;
         }
 
+        super.destroy();
```

Comment:
That doesn't work, this would be the result:

![image](https://github.com/playcanvas/engine/assets/5236548/1fab0ed7-0b98-4287-909c-35ebf8fc38d4)

Reason I described here: https://github.com/playcanvas/engine/pull/5414#issuecomment-1599076644

The components think they can remove entities from the hierarchy without deleting them, so the get orphaned basically and the recursion doesn't find them any longer.